docs/deploy_streamlit_app.md

```diff
@@ -1,3 +1,156 @@
+
+Of course, if you want to host your app using another hosting provider, go for it! Streamlit apps work anywhere a Python app works. You can find guides for other hosting providers on our [community-supported deployment wiki](https://discuss.streamlit.io/t/streamlit-deployment-guide-wiki/5099).
+
+## Get a Streamlit sharing account
+
+To get started, first request an invite at [streamlit.io/sharing](http://streamlit.io/sharing). Streamlit sharing is currently invitation-only while we ramp things up. We promise to get you one very quickly ❤️
+
+Once you have your invite you're ready to deploy! It's really straightforward, just follow the next few steps.
+
+## Put your Streamlit app on GitHub
+
+Make sure it's in a public folder and that you have a [requirements.txt file](https://pip.pypa.io/en/stable/reference/pip_install/#requirements-file-format)
+
+- If you need to generate a requirements file, try using `pipreqs`
+
+```bash
+pip install pipreqs
+pipreqs /home/project/location
+```
+
+```eval_rst
+.. note:: You should only include packages in requirements.txt that are not distributed with a standard Python installation (i.e. only packages that need to be installed from PyPI/conda). If you `include any of these modules from base Python <https://docs.python.org/3/py-modindex.html>`_ in your requirements.txt file, you will get an error when you try to deploy.
+```
+
+- If you have requirements for apt-get, add them to `packages.txt`, one package name per line. See our streamlit-apps demo repo for an [example packages.txt file](https://github.com/streamlit-apps/ml-projects/blob/master/packages.txt).
+
+## Log in to share.streamlit.io
+
+The first thing you'll see is a button to login with GitHub. Please login with the primary email associated with your GitHub account.
+
+```eval_rst
+.. important:: If the email you originally signed-up with isn't the primary email associated with your GitHub account, just reply to your invite email telling us your primary Github email so we can grant access to the correct account.
+```
+
+You can find your [GitHub account email](https://github.com/settings/emails) here:
+<img src="_static/img/GitHub_email.png" alt="Github email" width="450" style='display: block; margin-left: auto; margin-right: auto; padding-top: 15px;'/>
+
+## Deploy your app
+
+Click "New app", then fill in your repo, branch, and file path, and click "Deploy". Your app will take a minute or two to deploy and then you'll be ready to share!
+
+If your app has a lot of dependencies it may take a few minutes to deploy the first time. But after that, any change that does not touch your dependencies should show up immediately!
+
+That's it — you're done! Once your app is deployed it will be available at
```

Comment:
Lots of exclamation points on the 3 paragraphs above. Also repeating "minute" too much.

How about: 
```
Click "New app", then fill in your repo, branch, and file path, and click "Deploy". Your app will take a minute or two to deploy and then you'll be ready to share!

If your app has a lot of dependencies it may take a some time to deploy the first time. But after that, any change that does not touch your dependencies should show up immediately.

That's it — you're done! Once your app is deployed it will be available at
```